src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchLoader.java

```diff
@@ -0,0 +1,133 @@
+package com.intuit.graphql.orchestrator.batch;
+
+import com.intuit.graphql.orchestrator.ServiceProvider;
+import com.intuit.graphql.orchestrator.federation.EntityFetchingException;
+import com.intuit.graphql.orchestrator.federation.EntityQuery;
+import com.intuit.graphql.orchestrator.federation.metadata.FederationMetadata;
+import com.intuit.graphql.orchestrator.federation.metadata.KeyDirectiveMetadata;
+import graphql.GraphQLContext;
+import graphql.execution.DataFetcherResult;
+import graphql.introspection.Introspection;
+import graphql.language.Field;
+import graphql.language.InlineFragment;
+import graphql.language.SelectionSet;
+import graphql.language.TypeName;
+import graphql.schema.DataFetchingEnvironment;
+import org.apache.commons.collections4.CollectionUtils;
+import org.dataloader.BatchLoader;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static com.intuit.graphql.orchestrator.utils.IntrospectionUtil.__typenameField;
+
+public class EntityFetcherBatchLoader implements BatchLoader<DataFetchingEnvironment, DataFetcherResult<Object>> {
+
+    private final QueryResponseModifier queryResponseModifier = new DefaultQueryResponseModifier();
+    private final BatchResultTransformer batchResultTransformer;
+    private final String entityTypeName;
+    private final List<String> representationFieldTemplate;
+    private final ServiceProvider entityServiceProvider;
+
+    public EntityFetcherBatchLoader(FederationMetadata.EntityExtensionMetadata metadata, String fieldName) {
+        this.entityServiceProvider = metadata.getServiceProvider();
+        this.entityTypeName = metadata.getTypeName();
+        this.representationFieldTemplate = generateRepresentationTemplate(metadata, fieldName);
+        this.batchResultTransformer = new EntityFetcherBatchResultTransformer(metadata.getServiceProvider().getNameSpace(), metadata.getTypeName(), fieldName);
+    }
+
+    @Override
+    public CompletionStage<List<DataFetcherResult<Object>>> load(List<DataFetchingEnvironment> dataFetchingEnvironments) {
+        DataFetchingEnvironment dfeTemplate = dataFetchingEnvironments.get(0);
+        GraphQLContext graphQLContext = dfeTemplate.getContext();
+
+        List<Map<String, Object>> representations = dataFetchingEnvironments.stream()
```

Comment:
it is very important to maintain the same order of DFE and DFR. I don't know if stream will mess up your order. I might need to understand this better, as I do not know the representation query so well

Reply:
The representation queries are in the same order as the dfe. When dgs receives the representation queries, it is supposed to return the dfr in the same order. I attempted a few calls with multiple entity requests and it looked good.